src/backend/joanie/lms_handler/backends/openedx.py

```diff
@@ -93,21 +98,31 @@ def get_enrollment(self, username, resource_link):
         logger.error(response.content)
         return None
 
-    def set_enrollment(self, username, resource_link, active=True):
-        """Set enrollment for a user with a course run given its url."""
+    def set_enrollment(self, enrollment):
+        """Set enrollment for a user on the remote LMS using resource link as url."""
         base_url = self.configuration["BASE_URL"]
-        course_id = self.extract_course_id(resource_link)
+        course_id = self.extract_course_id(enrollment.course_run.resource_link)
+        mode = (
+            OPENEDX_MODE_VERIFIED
+            if enrollment.was_created_by_order
+            or enrollment.related_orders.filter(
+                state=enums.ORDER_STATE_VALIDATED
+            ).exists()
```

Comment:
Unfortunately I'm affraid this not work.

`enrollment.was_created_by_order` only means the enrollment has been created in the context of an order. But we could have enrollment that was not created in the context of an order but later included in an order. 

From my point of view, we should check if the enrollment user owns an order which target the enrollment course run.

Reply:
Good catch. So that's what I did now!